numba/np/random/generator_methods.py

```diff
@@ -819,3 +820,75 @@ def impl(inst, n, p , size=None):
                 out[i] = random_negative_binomial(inst.bit_generator, n, p)
             return out
         return impl
+
+
+@overload_method(types.NumPyRandomGeneratorType, 'noncentral_chisquare')
+def NumPyRandomGeneratorType_noncentral_chisquare(inst, df, nonc, size=None):
+    check_types(df, [types.Float, float], 'df')
+    check_types(nonc, [types.Float, float], 'nonc')
```

Comment:
The NumPy docs suggest that `df` and `nonc` can be an `array_like` of floats too. Having the restriction above is fine for an initial implementation but should be documented.

Reply:
NumPy also seems to "work" for types that will cast to float, e.g. `generator.noncentral_chisquare(12, 7, size=(3, 4))` works fine.

Reply:
Resolved in [9bcf3e2](https://github.com/numba/numba/pull/8520/commits/9bcf3e2edbdb4f1fdb5417ccada72d589b6da88e)